agent/app/agent.go

```diff
@@ -997,29 +999,67 @@ func (agent *ecsAgent) startACSSession(
 	taskHandler *eventhandler.TaskHandler,
 	doctor *doctor.Doctor) int {
 
-	acsSession := acshandler.NewSession(
-		agent.ctx,
-		agent.cfg,
-		deregisterInstanceEventStream,
-		agent.containerInstanceARN,
-		agent.credentialProvider,
-		agent.dockerClient,
+	inactiveInstanceCB := func() {
+		// If the instance is inactive (i.e., was deregistered), send an event to the event stream
+		// for the same.
+		err := deregisterInstanceEventStream.WriteToEventStream(struct{}{})
+		if err != nil {
+			seelog.Debugf("Failed to write to deregister container instance event stream, err: %v", err)
```

Comment:
Should we use logger for all new log statement so we have more structured logs? Is `debug` the common level for this type of error?

Reply:
> Should we use logger for all new log statement so we have more structured logs?

Sure, we can use logger defined in ecs-agent module for new log statements in this file. I will make this change in the next revision (assuming there are other blocking changes that need to be made). 

> Is debug the common level for this type of error?

Debug is the level that is currently being used for this scenario. See current corresponding code here:
https://github.com/aws/amazon-ecs-agent/blob/dev/agent/acs/handler/acs_handler.go#L186-L195